### PR TITLE
Make etc/ci/upload_nightly.sh responsible for putting datetimes into filenames

### LINKED ISSUE #12128: Make etc/ci/upload_nightly.sh responsible for putting datetimes into filenames
Part of #11980.

Currently, our Linux and Mac (although not Android) `./mach package` builders create files whose name include the date and time of building, which is a source of non-determinism. However, we do want to include the date and time for our nightly builds, so that multiple nightlies can coexist in a folder (e.g. `$HOME/downloads`) or S3 bucket (e.g. `servo-builds`). Two parts to this:
- Make `./mach package` output files that always use the same name (e.g. servo.tar.gz). File: `python/servo/package_commands.py`.
- Make `etc/ci/upload_nightly.sh`  insert the appropriate date and time when uploading the built nightly to S3. This depends on #12004 landing.

Bonus points (but not required for the E-easy): add a test that changing the date/time (e.g. with a date/time faking library) does not affect the filename or contents of the output.

### PR DESCRIPTION
**Pull request by imjacobclark**
This is part of #11980, allowing for more reproducible builds, increasing the security and integrity of the release process. 

This particular change ensures the date and time is on the files that are available for download from the nightly-release s3 bucket and removes that responsibility from the packager. 

See #12128 for more details.

@aneeshusa

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #12128 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because they are part of the CI infrastructure 

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12163)

### CODE REVIEW

**Comment by aneeshusa** [comment]
This should also incorporate the change in #12088.

File: etc/ci/upload_nightly.sh [hunk 1]
```diff
@@ -10,12 +10,12 @@ usage() {
     printf "usage: ${0} android|linux|mac|windows\n"
 }
 
-
 upload() {
-    local package_filename 
+    local package_filename current_date_time
```
Comment (aneeshusa) [inline etc/ci/upload_nightly.sh]: Let's rename `current_date_time` to `nightly_timestamp`.

**Comment by imjacobclark** [comment]
@aneeshusa Resolve the merge conflict and review comments, just need to wait on #12088 merging. 

Is #12088 still relevant in that file if this PR moves the time/date logic into `upload_nightly.sh`?

**Comment by aneeshusa** [comment]
Well, it just merged, so that's a moot point now :)

That PR changes the colons in the timestamp to dashes because the `tar` command will interpret the filename with colons as containing a remote server address with the colons and try to connect out, instead of knowing that it is a local file. I wish `tar` would just get smarter, but in the meantime we're switching to dashes so `tar` doesn't get confused. Any replacement for that functionality also needs to incorporate that change.

**Comment by imjacobclark** [comment]
@aneeshusa I will port the functionality from the Python into `upload_nightly.sh`, there seems to be a few conflicts on this PR too which I'll resolve shortly.

**Comment by aneeshusa** [comment]
#12088 merged so this is no longer blocked.

@imjacobclark are you still working on this? Do you have any questions?

**Comment by imjacobclark** [comment]
@aneeshusa still working on this, had a bit of a hectic few weeks, however will pick this up again now. Pretty straightforward, I'll catch up on IRC if there are any issues.

**Comment by KiChjang** [comment]
@imjacobclark Is this still being worked on?

**Comment by rjgoldsborough** [comment]
I'd like to help finish this one if it's up for grabs.

It looks like the datetime got moved to the upload_nightly.sh but still uses colons.
Was that the only outstanding change needed besides a rebase?

**Comment by aneeshusa** [comment]
@rjgoldsborough Yes, please feel free to submit a PR for this!

This needs a rebase, but if you look at `python/servo/package_commands.py` you'll see we replace colons by dashes, which is also behavior we need to keep.

**Comment by Wafflespeanut** [comment]
Closing based on inactivity

**Pull request closed without merging**